pkg/conversion/capi2mapi/powervs_test.go

```diff
@@ -0,0 +1,96 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("capi2mapi PowerVS", func() {
+
+	powerVSMachineTemplate := &capibmv1.IBMPowerVSMachineTemplate{
+		Spec: capibmv1.IBMPowerVSMachineTemplateSpec{
+			Template: capibmv1.IBMPowerVSMachineTemplateResource{
+				Spec: capibmv1.IBMPowerVSMachineSpec{
+					ImageRef:        &corev1.LocalObjectReference{Name: "rhcos-capi-powervs"},
+					ProviderID:      ptr.To("test-123"),
+					ServiceInstance: &capibmv1.IBMPowerVSResourceReference{Name: ptr.To("service-instance")},
+					Network:         capibmv1.IBMPowerVSResourceReference{Name: ptr.To("network")},
+				},
+			},
+		},
+		Status: capibmv1.IBMPowerVSMachineTemplateStatus{},
+	}
+
+	powerVSMachine := &capibmv1.IBMPowerVSMachine{
+		Spec: capibmv1.IBMPowerVSMachineSpec{
+			ImageRef:        &corev1.LocalObjectReference{Name: "rhcos-capi-powervs"},
+			ProviderID:      ptr.To("test-123"),
+			ServiceInstance: &capibmv1.IBMPowerVSResourceReference{Name: ptr.To("service-instance")},
+			Network:         capibmv1.IBMPowerVSResourceReference{Name: ptr.To("network")},
+		},
+	}
+
+	capiMachine := &capiv1.Machine{
+		Spec: capiv1.MachineSpec{
+			ClusterName: "test-123",
+		},
+	}
+
+	powerVSMachineSet := &capiv1.MachineSet{
+		Spec: capiv1.MachineSetSpec{
+			Replicas:    ptr.To(int32(2)),
+			ClusterName: "test-123",
+			Template: capiv1.MachineTemplateSpec{
+				Spec: capiv1.MachineSpec{
+					ClusterName: "test-123",
+				},
+			},
+		},
+	}
+
+	capiPowerVSCluster := &capibmv1.IBMPowerVSCluster{
+		Spec: capibmv1.IBMPowerVSClusterSpec{
+			ServiceInstance: &capibmv1.IBMPowerVSResourceReference{Name: ptr.To("serviceInstance")},
+			Zone:            ptr.To("test-zone"),
+		},
+		Status: capibmv1.IBMPowerVSClusterStatus{
+			Ready: true,
+		},
+	}
+
+	It("should be able to convert a Power VS CAPI Machine to a Power VS MAPI Machine", func() {
+		mapiMachine, warns, err :=
+			FromMachineAndPowerVSMachineAndPowerVSCluster(capiMachine, powerVSMachine, capiPowerVSCluster).ToMachine()
+		Expect(err).ToNot(HaveOccurred(), "should have been able to convert CAPI Machine/PowerVSMachineTemplate to MAPI Machine")
+		Expect(mapiMachine).To(Not(BeNil()), "should not have a nil MAPI Machine")
+		Expect(warns).To(BeEmpty(), "should have not warned while converting CAPI Machine/PowerVSMachineTemplate to MAPI Machine")
+	})
+
+	It("should be able to convert a Power VS MAPI MachineSet to a Power VS CAPI MachineSet", func() {
+		mapiMachineSet, warns, err :=
+			FromMachineSetAndPowerVSMachineTemplateAndPowerVSCluster(powerVSMachineSet, powerVSMachineTemplate, capiPowerVSCluster).ToMachineSet()
+		Expect(mapiMachineSet).To(Not(BeNil()), "should not have a nil MAPI MachineSet")
+		Expect(err).ToNot(HaveOccurred(), "should have been able to convert CAPI MachineSet/PowerVSMachineTemplate to MAPI MachineSet")
+		Expect(warns).To(BeEmpty(), "should have not warned while converting CAPI MachineSet/PowerVSMachineTemplate to MAPI MachineSet")
+	})
+})
```

Comment:
This has changed quite a bit since the first test. I suggest following this pattern: 
https://github.com/openshift/cluster-capi-operator/pull/222
https://github.com/openshift/cluster-capi-operator/pull/215

And maybe put together a resourcebuilder for CAPIBM, like so :)
https://github.com/openshift/cluster-api-actuator-pkg/pull/333

Reply:
Should I update the this PR with similar change or Shall I submit a follow up PR?

also I will submit a PR for adding resourcebulder for CAPIBM as well soon.

Reply:
I'd say let's keep it in one PR, to avoid conflict nerves and stuff :)

Reply:
Thanks @Karthik-K-N

Reply:
sure will do that

Reply:
Added more tests.